app/src/main/java/de/danoeh/antennapod/dialog/FeedsApplyActionFragment.java

```diff
@@ -0,0 +1,365 @@
+package de.danoeh.antennapod.dialog;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.ListView;
+
+import androidx.annotation.IdRes;
+import androidx.annotation.NonNull;
+import androidx.annotation.PluralsRes;
+import androidx.annotation.StringRes;
+import androidx.appcompat.app.AlertDialog;
+import androidx.appcompat.widget.Toolbar;
+
+import androidx.collection.ArrayMap;
+import androidx.fragment.app.Fragment;
+
+import com.google.android.material.snackbar.Snackbar;
+import com.leinardi.android.speeddial.SpeedDialView;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import de.danoeh.antennapod.R;
+import de.danoeh.antennapod.activity.MainActivity;
+import de.danoeh.antennapod.core.feed.Feed;
+import de.danoeh.antennapod.core.feed.FeedItem;
+import de.danoeh.antennapod.core.storage.DBWriter;
+import de.danoeh.antennapod.core.util.FeedItemPermutors;
+import de.danoeh.antennapod.core.util.FeedPermutors;
+import de.danoeh.antennapod.core.util.LongList;
+import de.danoeh.antennapod.core.util.FeedPermutors.SortOrder;
+import de.danoeh.antennapod.ui.common.ThemeUtils;
+import io.reactivex.Completable;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+
+@SuppressLint("NewApi")
+public class FeedsApplyActionFragment extends Fragment implements Toolbar.OnMenuItemClickListener {
+    private static final String TAG = "FeedsApplyActionFragmen";
+    public static final int ACTION_REMOVE_NEW_FLAGS = 1;
+    public static final int ACTION_MARK_AS_PLAYED = 2;
+    public static final int ACTION_REMOVE_FEED = 3;
+    public static final int ACTION_ALL = ACTION_REMOVE_NEW_FLAGS | ACTION_MARK_AS_PLAYED
```

Comment:
Please also add support for setting 'keep updated' and 'include in auto downloads' to fix #4681. Not sure how they could be included without adding a ton of buttons behind the FAB. Maybe one FAB button for 'keep updated' that then opens a dialog asking if you want to disable or enable?

Reply:
Sure, I can add them. @keunes had the idea of having one single fab that opens a modified podcast settings screen. I want to suggest a solution that combines both. Instead let's have the most important options as sub fabs and have an 'other' sub fab that instead of opening a modified podcasts settings screen, opens another dialog with the rest of the options displayed. That way we don't have a ton of mini fabs and we can reduce complexity by not having to 'modify' a podcast settings screen (which could confuse some users, even if there is a clarifying message).